### PR TITLE
Add u-boot v2018.07 recipe

### PR DESCRIPTION
**Pull request by ZubairLK**
u-boot v2018.07 has quite a few fixes in upstream that we would like
to be present. Add u-boot v2018.07 recipe and refactor patches

Tested on pi3b+ only.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Change log
 -----------
 
+* Add u-boot v2018.07 recipe and use that
```
Comment (floion) [inline CHANGELOG.md]: Add name here in `[]`

Comment (ZubairLK) [reply CHANGELOG.md]: Added. Thanks

File: layers/meta-resin-raspberrypi/recipes-bsp/u-boot/u-boot-common_2018.07.inc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+HOMEPAGE = "http://www.denx.de/wiki/U-Boot/WebHome"
```
Comment (agherzan) [inline layers/meta-resin-raspberrypi/recipes-bsp/u-boot/u-boot-common_2018.07.inc]: Please specify in the commit log the revision from which you imported this recipe.

File: layers/meta-resin-raspberrypi/recipes-bsp/u-boot/u-boot_%.bbappend [hunk 3]
```diff
@@ -3,7 +3,14 @@ UBOOT_KCONFIG_SUPPORT = "1"
 
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
+PREFERRED_VERSION_python = "2018.07"
```
Comment (agherzan) [inline layers/meta-resin-raspberrypi/recipes-bsp/u-boot/u-boot_%.bbappend]: Why do you need this? Do we have another set preferred version?

Comment (ZubairLK) [reply layers/meta-resin-raspberrypi/recipes-bsp/u-boot/u-boot_%.bbappend]: why does that say python lol. that should say PREFERRED_VERSION_u-boot!

lol

**Pull request merged**